src/routes/object/getObject.ts

```diff
@@ -27,8 +27,17 @@ const getObjectParamsSchema = {
   },
   required: ['bucketName', '*'],
 } as const
+
+const getObjectQuerySchema = {
+  type: 'object',
+  properties: {
+    filename: { type: 'string', example: 'filename.json' },
```

Comment:
What about supporting this via `response-content-disposition` and users can do `object/auth/cat.png?response-content-disposition=attachment`. It is more obvious than the `filename` parameter since `response-content-disposition` makes it explicit that it is to trigger the file as a download. It also matches the [S3 format](https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObject.html#API_GetObject_RequestSyntax). 

If users want to specify a different filename when downloading, which [turns out to be an issue](https://engineering.resolvergroup.com/2022/02/aws-s3-utf-8-content-disposition/#The%20Problem%20with%20UTF-8%20in%20the%20%E2%80%98response-content-disposition%E2%80%99%20Parameter) when we support unicode characters in the filename, they can also specify the filename using the same parameter like `object/auth/cat.png?response-content-disposition=attachment; filename=cat.png`

We will also need to add this parameter to our client libs

Reply:
this approach makes sense, my initial idea was based on the fact that `Content-Disposition` could only be:

- attachment
- inline (default)

however, as you are suggesting the file name could be Unicode based, so probably we could support this by default on the backend.

For example, an alternative more dev-friendly approach could be:

adding a `?download=1` query string parameter with a value of `1` would force the file to be downloaded with the original name.

Where else if the value is specified with any other strings, ex: `?download=file.jpg` or `?download=고양이.jpg` we would download the file with the given name.

I believe your proposed approach gives more freedom to clients to customize the header, whereas the second approach takes a more opinionated approach on how the header should be behaving simplifying the use.

What do you think?

Reply:
Ah that's interesting. Agree that we should prioritize DX than maintaining compatibility (the method I proposed is not completely compatible with S3's XML based API anyway). 

What about this? `?download` forces a download and users can optionally specify a different file name via `?download=newfilename`

Reply:
I like that!

New implementation changed, we can now download the file with:

- `?download` results in the original file name
- `?download=name.jpg` specify a name

it also supports UTF-8 chars, ex: `?download=고양이.jpg`

Reply:
Cross linking PR for `storage-js` https://github.com/supabase/storage-js/pull/112